### PR TITLE
Add sitemap

### PR DESCRIPTION
**Pull request by kloimhardt**
Adresses Issue #50.

- [X] Create sitemap in XML format
- [X] Add link to robots.txt
- [x] Prevent hardcoded URL https://clojurians-log.clojureverse.org
- [x] Decide whether or not to have a link to the sitemap XML on main page

### CODE REVIEW

**Comment by plexus** [comment]
Here's Google's documentation about site maps

https://support.google.com/webmasters/answer/183668?hl=en

**Comment by kloimhardt** [comment]
For some reason, my latest push to the add-sitemap branch does not show up here. After googling, it seems the reason is that I merged "About-in-table" (although to be honest I do not understand at all what is going on, as I later merged "master" just before opening the PR). One recommendation is to open a new PR (rebase seems not to be recommended for PRs in public repositories). Should we close this PR, or do you maybe have an idea where the problem could be, so I can tackle it in another way?

**Comment by plexus** [comment]
Not sure @kloimhardt, hard for me to see the state of your repo. Maybe try a graphical git client like gitg to see how the branches relate to each other.
 
> rebase seems not to be recommended for PRs in public repositories

I'm personally 100% ok with rebasing a PR like this where it's not merged anywhere, and you are the only person working on it.

But if it's easier to just create a new PR then that's fine too, whatever works. PRs are cheap :) just make sure to reference one from the other so it's clear what the situation is.

File: src/clojurians_log/views.clj [hunk 1]
```diff
@@ -247,10 +247,15 @@
...
+        [:div.app-title
+         [:a {:href "/"}
+          (get-in context [:request :clojurians-log.application/title])]]]
        [:td.padding-15px
         [:a {:href (path-for context :clojurians-log.routes/about)}
-         "About"]]]]
+         "About"]]
+       [:td.padding-15px
+        [:a {:href (path-for context :clojurians-log.routes/sitemap)}
+         "Sitemap"]]]]
```
Comment (kloimhardt) [inline src/clojurians_log/views.clj:258]: Do we need this link to the sitemap, or just have it in robots.txt?

File: src/clojurians_log/views.clj [hunk 2]
```diff
@@ -271,6 +276,18 @@
        (get-in context [:request :clojurians-log.application/title])]]
      (:data/about-hiccup context)]]])
 
+(defn- sitemap-xml [{:data/keys [channel-day-tuples] :as context}]
+  [:urlset {:xmlns "http://www.sitemaps.org/schemas/sitemap/0.9"}
+   (for [[{:channel/keys [name]} channel-days] channel-day-tuples]
+     (for [[day cnt] channel-days]
+       [:url
+        [:loc (str "https://clojurians-log.clojureverse.org"
```
Comment (kloimhardt) [inline src/clojurians_log/views.clj:284]: How can we avoid this hardcoded Url?

**Comment by kloimhardt** [comment]
Thank you **@plexus** for the clarification. Did rebase and force-push on existing PR. Can you review the code? At the moment the above two questions are open from my side, can you help me with those decisions?

File: src/clojurians_log/views.clj (hunk 1 above)
Comment (plexus) [reply src/clojurians_log/views.clj:258]: Doesn't hurt to have it, but yes it also should go into robots.txt, which I guess we don't check in here.

File: src/clojurians_log/views.clj (hunk 2 above)
Comment (plexus) [reply src/clojurians_log/views.clj:284]: There should be a `:data/http-origin` in the context map that you can use.

File: src/clojurians_log/views.clj (hunk 1 above)
Comment (kloimhardt) [reply src/clojurians_log/views.clj:258]: So I leave the link here. I amended the file resources/public/robots.txt as well.

**Comment by kloimhardt** [comment]
PR is ready for merging into master.

**Comment by plexus** [comment]
ship it squirrel

**Pull request merged**